sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -583,7 +583,9 @@ object FunctionRegistry {
           val validParametersCount = constructors
             .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
             .map(_.getParameterCount).distinct.sorted
-          val expectedNumberOfParameters = if (validParametersCount.length == 1) {
+          val expectedNumberOfParameters = if (validParametersCount.isEmpty) {
+            constructors.headOption.map(_.getParameterCount).getOrElse(0).toString
```

Comment:
use headOption? This is hacky. We should follow the same logic what we are doing here.

Reply:
The problem is that we have function registered `in` but that's unable/not supposed to to call via `in(...)`. Possibly there might be more cases like this. Ideally we might have to identify and remove those expressions out but wanted to narrow the scope this PR by such change.